### PR TITLE
Introduce new 'vhostless' flow

### PR DESCRIPTION
**Pull request by jaeaster**
Used for returning multiple tokens baed on multiple authenticated users.

Not mean to be exposed externally. Rather this flow is set by doorkeeper when vhost is empty.

### CODE REVIEW

File: lib/doorkeeper/request/vhostless.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+require 'doorkeeper/request/strategy'
+
+module Doorkeeper
+  module Request
+    class Vhostless < Strategy
+      delegate :credentials, :resource_owners, :parameters, to: :server
+
+      
+
```
Comment (markabishop) [inline lib/doorkeeper/request/vhostless.rb]: minor lots of extra lines

**Review by markabishop — APPROVED** [review]
minor nit on extra lines

File: lib/doorkeeper/oauth/vhostless_access_token_request.rb [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        find_or_create_access_tokens(client, resource_owners, scopes, server)
+      end
+
+      def find_or_create_access_tokens(client, resource_owners, scopes, server)
+        @access_tokens = resource_owners.map do |resource_owner|
+          AccessToken.find_or_create_for(
+            client,
+            resource_owner,
+            scopes,
+            Authorization::Token.access_token_expires_in(server, client),
```
Comment (markabishop) [inline lib/doorkeeper/oauth/vhostless_access_token_request.rb:45]: Is it the same scope and server for all resource_owners? arent those different?

Comment (jaeaster) [reply lib/doorkeeper/oauth/vhostless_access_token_request.rb:45]: No it's not - scopes are passed in as params. Server means something different here, not resource server

File: lib/doorkeeper/request/vhostless.rb (hunk 1 above)
Comment (jaeaster) [reply lib/doorkeeper/request/vhostless.rb]: Fixed

**Pull request merged**